### PR TITLE
test(cli): Track --help output

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

This makes it easier to evaluate the usability of PRs, like #11905

This follows the pattern of `cargo add` and `cargo remove` of putting these ui tests in `cargo_<cmd>/` directories.  `init` didn't follow this pattern, so it was renamed to `cargo_init/`.  `cargo_config.rs` was going to conflict with this, it was merged in.

We can evaluate other `<cmd>.rs` files at a later point and consolidate.

### How should we test and review this PR?

The main risks are
- Are all files linked together (`main.rs` -> `<cmd>/mod.rs` -> `<cmd>/<help>.rs`
- Are all `help/mod.rs`s pointing to the right command

### CODE REVIEW

**Comment by ehuss** [comment]
I'm a little concerned that this introduces two modules for almost every command, and it may not be clear how to manage tests with that structure. That is, there is a `yank.rs` and a `cargo_yank/mod.rs`, and it may not be clear why that separation is made, or guidance on how to organize tests. Is there some way we can avoid that separation? What about moving the corresponding `yank.rs` to `cargo_yank/mod.rs`?

File: tests/testsuite/cargo/help/mod.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+use cargo_test_support::curr_dir;
```
Comment (ehuss) [inline tests/testsuite/cargo/help/mod.rs:1]: Just curious, is there a reason `curr_dir` is not in `cargo_test_support::prelude`?

Comment (epage) [reply tests/testsuite/cargo/help/mod.rs:1]: Not really. I can add it if wanted.

**Comment by epage** [comment]
> I'm a little concerned that this introduces two modules for almost every command, and it may not be clear how to manage tests with that structure. That is, there is a yank.rs and a cargo_yank/mod.rs, and it may not be clear why that separation is made, or guidance on how to organize tests. Is there some way we can avoid that separation? What about moving the corresponding yank.rs to cargo_yank/mod.rs?

I had originally wanted to punt on `yank.rs` vs `cargo_yank.rs` for now and called this out in the PR description.  My assumption in doing so is there would be some untangling of relationships that might not be obvious.  Also my primary goal is implementing  #11905. and I didn't want to get too distracted with other steps along the way.

File: tests/testsuite/cargo/help/mod.rs (hunk 1 above)
Comment (ehuss) [reply tests/testsuite/cargo/help/mod.rs:1]: That might help with some of the boilerplate. It looks like it is used for essentially every UI test. This is unrelated to this PR, though, and can be done any time.

**Comment by ehuss** [comment]
If you want to defer on figuring out the module structure, would it be possible to just add the tests to the existing modules in the mean time? I assume `stdout_matches_path` can point to some directory with a more manual path (like `curr_dir!().join("cargo_yank/help/stdout.log")`)?

I would prefer to avoid moving forward with something that puts it in an incomplete state without some idea or strategy for where it is intended to end up.

**Comment by epage** [comment]
> If you want to defer on figuring out the module structure, would it be possible to just add the tests to the existing modules in the mean time? I assume stdout_matches_path can point to some directory with a more manual path (like curr_dir!().join("cargo_yank/help/stdout.log"))?

This does not get us out of being in an inconsistent, intermediate state since this would then make the UI tests inconsistent with each other.  Since those all follow a very similar pattern, I would like to keep them as such for easily modifying them in the future.

> I would prefer to avoid moving forward with something that puts it in an incomplete state without some idea or strategy for where it is intended to end up.

Except thats what software evolution is, is to always be in some kind of incomplete state as you work towards your ideal.  To take a stance you can't improve anything until everything is improved leaves you always in a worse place than small, incremental progress.

This was meant to be the simple, quick PR to unblock other help work.  I feel like these discussions are already exceeding a reasonable time box for that scope but stalls out future improvements in other areas.

**Comment by epage** [comment]
Discussion on test organization has moved to https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Brainstorming.20on.20test.20organization with the consent of @ehuss

**Comment by epage** [comment]
@bors r=ehuss

With the above Zulip discussion, @ehuss told me to merge this once it was ready

**Pull request merged**